recipes/cfcTest.tsx

```diff
@@ -0,0 +1,95 @@
+import { h } from "@commontools/html";
+import {
+  derive,
+  handler,
+  JSONSchema,
+  NAME,
+  recipe,
+  str,
+  UI,
+} from "@commontools/builder";
+
+const UpdaterSchema = {
+  type: "object",
+  properties: {
+    newValues: {
+      type: "array",
+      items: {
+        type: "string",
+        $ifc: { classification: ["confidential"] },
+      },
+    },
+  },
+  title: "Update Values",
+  description: "Append `newValues` to the list.",
+  example: { newValues: ["foo", "bar"] },
+  default: { newValues: [] },
+} as const satisfies JSONSchema;
+
+const InputSchema = {
+  type: "object",
+  properties: {
+    values: {
+      type: "array",
+      items: { type: "string", $ifc: { classification: ["confidential"] } },
+      asCell: true,
+    },
+  },
+  default: {
+    values: ["string"],
+    $ifc: { classification: ["unclassified"], integrity: ["commontools"] },
+  },
+} as const satisfies JSONSchema;
+
+const OutputSchema = {
+  type: "object",
+  properties: {
+    values: { type: "array", items: { type: "string" } },
+    updater: {
+      asStream: true,
+      ...UpdaterSchema,
+    },
+  },
+} as const satisfies JSONSchema;
+
+const updater = handler(
+  UpdaterSchema,
+  InputSchema,
+  (event, state) => {
+    console.log("updating values", event);
+    event.newValues.forEach((value) => {
+      console.log("adding value", value);
+      state.values.push(value);
+    });
+  },
+);
+
+const adder = handler({}, InputSchema, (_, state) => {
```

Comment:
we didn't discuss this before, but what's interesting here, is that the handler schema isn't just an input (as in "just reading") schema, but also passes in the cells that are used to write to, so for `handler` a single schema captures both additional state to read (in addition to the event) as well as where outputs are written to. for `lift` that's technically also allowed today, but we could constrain that to just reads and just writes in the input and output schemas.

come to think of it, the return value of `handler` can be a recipe itself, and we could eventually add a schema to restrict its shape. for later.